prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -79,7 +80,8 @@ object FHIRBundleHelpers {
         val provenanceResource = try {
             fhirBundle.entry.first { it.resource.resourceType.name == "Provenance" }.resource as Provenance
         } catch (e: NoSuchElementException) {
-            throw IllegalStateException("The FHIR bundle does not contain a Provenance resource")
+            fhirBundle.addEntry().resource = Provenance()
+            fhirBundle.entry.first { it.resource.resourceType.name == "Provenance" }.resource as Provenance
```

Comment:
Looks like this change breaks test `manipulating bundle without provenance`, that might not have existed in the other branch. Just make sure you can run `test` and `testIntegration` locally to make sure there aren't other cases we missed here.